pkg/build/strategies/sti/sti.go

```diff
@@ -252,35 +275,45 @@ func (builder *STI) Prepare(config *api.Config) error {
 		// we're validating values here to be sure that we're handling both of the cases of the invocation:
 		// from main() and as a method from OpenShift
 		for _, volumeSpec := range builder.config.RuntimeArtifacts {
-			if !path.IsAbs(volumeSpec.Source) {
-				return fmt.Errorf("Invalid runtime artifacts mapping: %q -> %q: source must be an absolute path", volumeSpec.Source, volumeSpec.Destination)
-			}
-			if path.IsAbs(volumeSpec.Destination) {
-				return fmt.Errorf("Invalid runtime artifacts mapping: %q -> %q: destination must be a relative path", volumeSpec.Source, volumeSpec.Destination)
+			var volumeErr error
+
+			switch {
+			case !path.IsAbs(volumeSpec.Source):
+				volumeErr = fmt.Errorf("Invalid runtime artifacts mapping: %q -> %q: source must be an absolute path", volumeSpec.Source, volumeSpec.Destination)
+			case path.IsAbs(volumeSpec.Destination):
+				volumeErr = fmt.Errorf("Invalid runtime artifacts mapping: %q -> %q: destination must be a relative path", volumeSpec.Source, volumeSpec.Destination)
+			case strings.HasPrefix(volumeSpec.Destination, ".."):
+				volumeErr = fmt.Errorf("Invalid runtime artifacts mapping: %q -> %q: destination cannot start with '..'", volumeSpec.Source, volumeSpec.Destination)
```

Comment:
i've reproduced this because i'm confused. https://play.golang.org/p/XkZR8jcOLW 
a switch statement continues till it hits a true case or a default if it doesn't? 
by the way, the first check is against volumeSpec.Source and the next two are for volumeSpec.Destination in case you think the first is against the same?